pyi.py

```diff
@@ -292,10 +292,15 @@ def _check_subscript_version_check(self, node: ast.Compare) -> None:
                 # Python 3.9 flattens the AST and removes Index, so simulate that here
                 slice_num = slc if isinstance(slc, ast.Num) else slc.value
                 # anything other than the integer 0 doesn't make much sense
-                if isinstance(slice_num, ast.Num) and slice_num.n == 0:
+                if (
+                    isinstance(slice_num, ast.Num)
+                    and isinstance(slice_num.n, int)
+                    and slice_num.n == 0
+                ):
                     must_be_single = True
                 else:
                     self.error(node, Y003)
+                    return
             elif isinstance(slc, ast.Slice):
                 if slc.lower is not None or slc.step is not None:
                     self.error(node, Y003)
```

Comment:
```suggestion
                    return
            elif isinstance(slc, ast.Slice):
                if slc.lower is not None or slc.step is not None:
                    self.error(node, Y003)
                    return
```
Good catch, missed this one. I don't think multiple errors are helpful because the text is the same for each of them.